bundles/org.eclipse.core.databinding.property/src/org/eclipse/core/internal/databinding/property/value/MapDelegatingValueObservableMap.java

```diff
@@ -162,36 +162,34 @@ private MapDiff<K, V> convertDiff(MapDiff<? extends K, ? extends I> diff) {
 			Map<K, V> newValues = new HashMap<>();
 
 			Set<? extends K> addedKeys = diff.getAddedKeys();
-			for (K key : addedKeys) {
+			addedKeys.forEach(key -> {
```

Comment:
Can you explain why this style is better the a classic foreach?

Also the commit message do not reflect this well as this is not an iterator (and I can't see iterators are legacy).

Reply:
my understanding is `lambda` is preferred over traditional iterations(with trivial `for` loops for explicit `collection` iterators). Let me know if that's not the case., and I can remove this change.